src/inewsStoryParser.ts

```diff
@@ -0,0 +1,154 @@
+				tapeTime: { value: undefined },
+				audioTime: { value: undefined },
+				totalTime: { value: undefined },
+				cumeTime: { value: undefined },
+				backTime: { value: undefined },
+				pageNumber: { value: undefined },
+				layout: { value: undefined },
+				runsTime: { value: undefined },
+				videoId: { value: undefined },
+			},
+			meta: {},
+			cues: [],
+		}
+
+		nodes.forEach(function (node) {
+			parseNsmlNode(node, story)
+		})
+
+		return story as INewsStory
+	}
+
+	function stringifyNodes(nodes: htmlparser.DOM): string {
+		let nodeStr = ''
+		if (Array.isArray(nodes)) {
+			nodes.forEach((node) => {
+				nodeStr += stringifyNode(node)
+			})
+		}
+		return nodeStr
+	}
+
+	function stringifyNode(node: htmlparser.Node): string {
+		let nodeStr = ''
+		if (node.type === 'text') return node.data
+		else if (node.type === 'tag') {
+			nodeStr = '<' + node.name
+			const attrStr = stringifyAttributes(node.attribs)
+			if (attrStr.length > 0) nodeStr += ' ' + attrStr
+			nodeStr += '>'
+			nodeStr += stringifyNodes(node.children)
+			nodeStr += '</' + node.name + '>'
+		}
+
+		return nodeStr
+	}
+
+	function stringifyAttributes(attributes: htmlparser.NodeAttributes): string {
+		let attrStr = ''
+		for (const key in attributes) {
+			if (attrStr.length > 0) attrStr += ' '
+			attrStr += key + '="' + attributes[key].replace(/"/g, '\\"') + '"'
+		}
+		return attrStr
+	}
+
+	function nodesToArray(nodes: htmlparser.DOM, tag: string) {
+		let lines: string[] = []
+
+		nodes.forEach((node) => {
+			if (node.type === 'tag') {
+				if (!node.children) {
+					return
+				}
+				if (node.name === tag) {
+					lines.push(unescape(stringifyNodes(node.children)))
+				}
+				lines = lines.concat(nodesToArray(node.children, tag))
+			}
+		})
+
+		// Filter out leading lines in production cues
+		lines = lines.filter((line) => {
+			return (line as any) > 0 || line != ']] S3.0 G 0 [['
+		})
+
+		return lines
+	}
+
+	function parseNsmlNode(node: htmlparser.Node, story: Partial<INewsStory>) {
+		if (node.type === 'tag') {
+			switch (node.name) {
+				case 'ae':
+					try {
+						const id = parseInt(node.attribs['id'], 10)
+						// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+						story.cues![id] = nodesToArray(node.children, 'ap')
+					} catch (error) {
+						// Do nothing.
+					}
+					break
+				case 'body':
+					story.body = unescape(stringifyNodes(node.children))
+					break
+				case 'meta':
+					story.meta = node.attribs
+					break
+				case 'storyid':
+					try {
+						story.id = node.children[0]['data']
+					} catch (error) {
+						// Do nothing.
+					}
+					break
+				case 'f':
+					try {
+						const key = node.attribs['id']
+						const val = node.children[0]['data']
+						const uec = Boolean(node.attribs['uec'])
+						const urgency = (
+							'urgency' in node.attribs ? parseInt(node.attribs['urgency'], 10) : undefined
+						) as INewsFieldAttributes['urgency']
```

Comment:
Instead of casting to the desired type, perhaps sanitizing the urgency value would be a better option. I have provided a suggestion for the implementation, such that the can be changed to:
```suggestion
						const urgency = sanitizeUrgency(node.attribs['urgency'])
```
by using
```typescript
function sanitizeUrgency(unsanitizedUrgency: string | undefined): INewsFieldAttributes['urgency'] {
	if (unsanitizedUrgency === undefined) {
		return
	}
	const urgency = parseInt(unsanitizedUrgency, 10)
	return isValidUrgency(urgency) ? urgency : undefined
}

function isValidUrgency(urgency: number): urgency is NonNullable<INewsFieldAttributes['urgency']> {
	return urgency >= 1 && urgency <= 3
}
```